### PR TITLE
More SkyCoord benchmarks

### PR DESCRIPTION
**Pull request by adrn**
This tests different paths through the SkyCoord initializer in a bit more of a controlled manner.

### CODE REVIEW

**Comment by adrn** [comment]
Note this also addresses astropy/astropy#7521

**Comment by adrn** [comment]
Ah, yes, that was a bug! But I don't know what could be causing the build failures. @astrofrog ?

File: benchmarks/coordinates.py [hunk 1]
```diff
@@ -57,14 +61,47 @@ def setup(self):
         self.array_ra = np.linspace(0., 360., 1000) * u.deg
         self.array_dec = np.linspace(-90., 90., 1000) * u.deg
 
-        np.random.seed(12345)
         self.icrs_scalar = ICRS(ra=1*u.deg, dec=2*u.deg)
-        self.icrs_array = ICRS(ra=np.random.random(10000)*u.deg,
-                               dec=np.random.random(10000)*u.deg)
+        self.icrs_array = ICRS(ra=np.ones(10000)*u.deg,
+                               dec=np.ones(10000)*u.deg)
```
Comment (astrofrog) [inline benchmarks/coordinates.py]: Why the change to just ones? Isn't it better to have a variety of values?

**Comment by astrofrog** [comment]
Just to clarify my previous comment - if you really want to modify a benchmark, that's of course fine, but you should rename it at the same time to make sure we don't have to go back and remove old results.

**Comment by eteq** [comment]
Using `asv dev --bench="coordinates.*"` I discovered some bugs in the benchmarks (note, @adrn, that seems to be the easiest way to debug the benchmarks themselves).  Note sure if that's the same as the CI failure though, so I pushed a commit up really just to see what happens...

I didn't address @astrofrog's suggestion of not modifying, though - will leave that to @adrn.

**Comment by adrn** [comment]
The benchmarks run for me locally with `asv dev`. I think I addressed @astrofrog 's comments by backing out the random seed changes. But I did rename some benchmarks for clarity.

**Comment by astrofrog** [comment]
The failure is in master and is unrelated
